internal/cssc/cssc.go

```diff
@@ -24,19 +27,42 @@ import (
 	"oras.land/oras-go/v2/registry/remote/retry"
 )
 
+type TagConvention string
+
+const (
+	Incremental TagConvention = "incremental"
+	Floating    TagConvention = "floating"
+)
+
+func (tc TagConvention) IsValid() error {
+	switch tc {
+	case Incremental, Floating:
+		return nil
+	}
+	return errors.New("tag-convention should be either incremental or floating")
+}
+
+var (
+	floatingTagRegex    = regexp.MustCompile(`^(.+)-patched`)             // Matches strings ending with -patched
+	incrementalTagRegex = regexp.MustCompile(`^(.+)-([1-9]\d{0,2})$`)     // Matches strings ending with -1 to -999
+	reservedSuffixRegex = regexp.MustCompile(`(-[1-9]\d{0,2}|-patched)$`) // Matches strings ending with -1 to -999 or -patched
```

Comment:
All of the other regexes include the prefix portion, namely ^(.+), do you need to match a full string for this one?